### PR TITLE
Add apply task option

### PR DESCRIPTION
**Pull request by tanmay17268**

### CODE REVIEW

File: project/src/main/java/com/google/sps/servlets/ApplyTask.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.ServletException;
+
+/** Servlet facilitating viewing task details. */
+@WebServlet("/task/apply")
+public class ApplyTask extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String taskId = getParameter(request, "taskId", "");
```
Comment (amanCoder110599) [inline project/src/main/java/com/google/sps/servlets/ApplyTask.java:32]: taskId should be a long datatype.

File: project/src/main/java/com/google/sps/servlets/ApplyTask.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+}
```
Comment (amanCoder110599) [inline project/src/main/java/com/google/sps/servlets/ApplyTask.java:69]: nit: newline at EOF

File: project/src/main/webapp/task.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      taskContainer.innerHTML = '<h1>' + response["title"] + '</h1>';
+      taskContainer.innerHTML += '<input type="text" name="taskId" value="' + queryParams.get("taskId") + '" style="display: none">';
+      taskContainer.innerHTML += '<p>' + response["details"] + "</p>";
+      taskContainer.innerHTML += '<p><strong>Location</strong>: ' + response["address"] + "</p>";
+      var deadline = new Date(response["deadline"]).toUTCString();
+      taskContainer.innerHTML += '<p><strong>Task Deadline</strong>: ' + deadline + "</p>";
+      taskContainer.innerHTML += '<p><strong>Created By</strong>: ' + response["creatorId"] + "</p>";
+      taskContainer.innerHTML += '<p><strong>Compensation</strong>: ' + response["compensation"] + "</p>";
+  })
+}
```
Comment (amanCoder110599) [inline project/src/main/webapp/task.js:16]: nit: newline at EOF

File: project/task_view_script.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+      const taskContainer = document.getElementById('task-container');
+      taskContainer.innerHTML = '<h1>' + response["title"] + '</h1>';
+      taskContainer.innerHTML += '<p>' + response["details"] + "</p>";
+      taskContainer.innerHTML += '<p><strong>Location</strong>: ' + response["address"] + "</p>";
+      var deadline = new Date(response["deadline"]).toUTCString();
+      taskContainer.innerHTML += '<p><strong>Task Deadline</strong>: ' + deadline + "</p>";
+      taskContainer.innerHTML += '<p><strong>Created By</strong>: ' + response["creatorId"] + "</p>";
+      taskContainer.innerHTML += '<p><strong>Compensation</strong>: ' + response["compensation"] + "</p>";
+  })
+}
```
Comment (amanCoder110599) [inline project/task_view_script.js:16]: nit: newline at EOF

File: project/task_view_script.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+function viewTaskDetails() {
+  var queryParams = new URLSearchParams(window.location.search);
+  queryParams = queryParams.toString();
+  fetch('/task?' + queryParams).then((response) => {
+      return response.json();
+  }).then((response) => {
+      const taskContainer = document.getElementById('task-container');
+      taskContainer.innerHTML = '<h1>' + response["title"] + '</h1>';
```
Comment (TheArijitSaha) [inline project/task_view_script.js:8]: You can access `response["title"]` as `response.title` too.

**Pull request merged**